server/embed/serve.go

```diff
@@ -99,7 +100,12 @@ func (sctx *serveCtx) serve(
 	splitHttp bool,
 	gopts ...grpc.ServerOption) (err error) {
 	logger := defaultLog.New(io.Discard, "etcdhttp", 0)
-	<-s.ReadyNotify()
+
+	select {
+	case <-s.ReadyNotify(): // wait for e.Server to join the cluster
+	case <-s.StopNotify(): // publish aborted from 'ErrStopped'
+		return errors.ErrStopped
```

Comment:
before the return `close(sctx.serversC)` needs to be invoked.
in my local test that resolves the issue.

Reply:
Good catch & updated.